Styling for cards on index page

trisalar opened the pull request:
I followed the suggested layout for the cards shown for issue #3 and did the following:
- Edited the structure in _talk.html.erb
  - the previous "votes" text and "+1" button is now a thumbs up button with a badge showing the total number of votes
  - the previous "assigned" link is now a button on the bottom right with the same functionality
- Edited main.scss
  - kept original ice class as that is also used in show.html.erb
  - new styles declared for index.html.erb in ice-index class

Screenshot of the new index cards:
![indexcardsample](https://cloud.githubusercontent.com/assets/7232696/10668657/e7ebe05e-790f-11e5-818e-7c2c0dc70e38.png)

trisalar committed:
Edit .ice and .ice a:hover styles in Custom CSS section

trisalar committed:
Create votes badge and assigned buttons in _talk.html.erb, add styles to main.scss for easier content alignment for each talk card

trisalar committed:
Edit _talk.html.erb so that bottom right 'assigned/unassigned' button is now a link with actual assignment functionality; remove redundant code in main.scss, index.html.erb now has its own ice-index class so that it won't clash with the styles for show.html.erb

colinrubbert commented:
I like the path you are going with this but it still needs work. Here are somethings we need to account for (I'll actually go and make comments at the code line so you know where things should be adjusted):
- On the `ice-index` `overflow: scroll;` shouldn't be there. It doesn't look good on the large devices so we should find an alternative way to deal with overflow issues such as adjusting the height of the div to accommodate extra text. Everything will eventually be static we just need to get the layout down and we'll be solid
- On mobile devices height should be set to auto. As previously stated height is hard set to make things look uniform in size on large displays, however, once it gets to a certain screen size height is no longer an issues size they are stacked on top of each other not side-by-side.
- Lets change `unscheduled talks`, `previous talks`, and `upcoming talks` div to be `div class="col-xs-12 col-md-10 col-md-offset-1` this will make it look better on mobile devices and not compromise the flow of the CSS
- In the cards let change the buttons div for mobile, right now they are stacked side by side and that causes things to overflow and look poorly on mobile. Lets change those to `div class="col-xs-12"` and then add a break after each button on mobile only and that will make it look a lot better. 

Overall it looks really good, I like the style and I like the direction we just need to polish it up a little bit more and make it look great for both desktops, laptops, tablets, and phones. Like I said before I will actually comment on the actual code commit so you know where I'm talking about the changes that need to be changed.

colinrubbert commented on app/assets/stylesheets/main.scss:
```diff
@@ -62,6 +63,40 @@ body {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  overflow: scroll;
```
We should remove this because it looks poorly having the scroll bars, we can remedy overflow in alternate ways to keep things looking sharp.

colinrubbert commented on app/assets/stylesheets/main.scss:
```diff
@@ -62,6 +63,40 @@ body {
+  flex-direction: column;
+  justify-content: space-between;
+  overflow: scroll;
+  height: 350px;
```
This should be set to auto for mobile devices with a media query.

``` CSS
@media (max-width 468px) {
height: auto;
}
```

Start with the `468px` breakpoint (typical breakpoint for cell phones) and work from there. We may want to carry this up higher to make it a little bit nicer on tablets but I have not tested what it looks like on tablets yet. This will be a good starting point though.

colinrubbert commented on app/views/talks/_talk.html.erb:
```diff
@@ -1,25 +1,45 @@
+
+  <div class="row">
+    <hr class="custom-hr" />
+    <div class="col-xs-6 col-md-6 text-center">
```
Lets change this to `div class="col-xs-12 col-md-6 text-center">`
This will make it look a lot better on mobile. Also add a breakpoint after this so that there is a space between the buttons but only on mobile. You can do this by doing something like this: `<br class="hidden-md hidden-lg">` might want to also add `hidden-sm` but see what this looks like first and go from there.

colinrubbert commented on app/views/talks/_talk.html.erb:
```diff
@@ -1,25 +1,45 @@
+      <% end %>
+    </div>
+
+    <div class="col-xs-6 col-md-6 text-center">    
```
Same here lets change this to `div class="col-xs-12 col-md-6 text-center">` to make it look better on mobile.

colinrubbert commented:
Let's change the divs on the unscheduled, scheduled, and previous talks to `<div class="col-xs-12 col-md-10 col-md-offset-1">`
Here are the respective locations of what we need to change so it'll be easier to hunt down on the code:
- https://github.com/FirehoseCommunity/LightningTalkTopics/blob/master/app/views/partials/_previousTalks.html.erb#L1
- https://github.com/FirehoseCommunity/LightningTalkTopics/blob/master/app/views/partials/_scheduledTalks.html.erb#L1
- https://github.com/FirehoseCommunity/LightningTalkTopics/blob/master/app/views/partials/_unscheduledTalks.html.erb#L1

trisalar commented:
Thanks for the feedback, Colin. I will definitely continue tackling this issue until we get it right!

After some research and much thought, I think removing the flexbox properties might be better due to browser support. The main reason I used it in the first place is because it made it a lot easier to align content vertically (i.e. for the buttons to stick to the bottom of the card). I will try to find other ways to accomplish the same effect, hopefully using CSS alone.

colinrubbert commented:
Absolutely, this is all part of the process of seeing things for all the angles. More you do this and the more feedback you get the more well rounded developer you will become. I really like how it's looking so I would love to see it continue down that road. The design really polishes things up. Keep it up, looking forward to seeing how it comes along.

trisalar committed:
Adjust bootstrap grid widths; buttons now stack on top of each other for xs viewport sizes

trisalar committed:
Change talk topic to h2 tag in _talk.html.erb, remove flexbox properties for browser support reasons, add/edit styles for site responsiveness

trisalar committed:
Remove redundant bootstrap class

trisalar committed:
Add 'align-bottom' class to div containing buttons, add js file that adjusts the height of the index cards and stickies the buttons to the bottom of each card

trisalar commented:
Here are a few adjustments I made to the CSS:
- adjusted the bootstrap col widths as suggested and removed overflow: scroll
- added col-sm to the div with ice-index class to cover for tablets (two cards per row)
- removed flexbox properties
- buttons now stack on top of each other for xs size viewports
- button size is reduced to 75% for smaller screen widths (when one card per row)

Design is a weakness of mine (trying to improve), but I thought that it might be easier to view the cards if they were aligned sided by side rather than like Pinterest since the cards don't contain any images. Thus said, I tried several ways to get the cards to align properly purely in CSS, but everything I tried was of no avail.

I ended up creating a JS file (index_cards.js) to help with the styling and functionality:
- when the index page is loaded, the height of the largest card (containing most content) is applied to all cards for horizontal alignment
- when the browser is resized, the height property of each card is reverted to the default of auto to take care of overflow problems during resizing
- the height of each card is readjusted for horizontal alignment again after browser resizing event ends

I left the overall design of the cards the same for now, and mainly dealt with the alignment of the elements. As always, feedback is most welcome!

colinrubbert commented:
Looking pretty good, I'm going to test it a little bit more and make sure there isn't any bugs in it but I like it.

trisalar committed:
Cache jQuery objects for better performance

trisalar commented:
Thanks, Colin.

Not sure how much of a performance gain it'll be, but upon looking it over again I realized I should've cached the jQuery objects so I just updated the file again.

colinrubbert commented:
Just merged into master. Great work on this!

colinrubbert commented:
I did just notice an issue on mobile, it still tries to resize the cards and it's a bit wonky. I think we should work on disabling the resizing on mobile and just set the height to auto.

trisalar commented:
Thanks, Colin. I just realized the index cards seem "jumpy" on iPads as well.

I will try to find a solution for both mobile phones and tablets, make adjustments, and submit the changes soon.
